site/public/js/calendar.js

```diff
@@ -0,0 +1,195 @@
+/**
+ * Gets the previous month of a given month
+ * @param month : int the current month (1 as January and 12 as December)
+ * @param year : int the current year
+ * @returns {number[]} : array<int> {previous_month, year_of_previous_month}
+ */
+// eslint-disable-next-line no-unused-vars
+function prevMonth(month, year) {
+    month = month - 1;
+    if (month <= 0) {
+        month = 12 + month;
+        year = year - 1;
+    }
+    return [month, year];
+}
+
+/**
+ * Gets the next month of a given month
+ * @param month : int the current month (1 as January and 12 as December)
+ * @param year : int the current year
+ * @returns {number[]} : array<int> {next_month, year_of_next_month}
+ */
+// eslint-disable-next-line no-unused-vars
+function nextMonth(month, year) {
+    month = month + 1;
+    if (month > 12) {
+        month = month - 12;
+        year = year + 1;
+    }
+    return [month, year];
+}
+
+/**
+ * Creates a HTML table cell that contains a date.
+ *
+ * @param year : int the year of the date
+ * @param month : int the month of the date (1 as January and 12 as December)
+ * @param day : int the date of the date (1 - 31)
+ * @param curr_view_month : int the current month that the calendar is viewing
+ * @returns {string} the HTML string containing the cell
+ */
+function displayDayCell(year, month, day, curr_view_month) {
+    const cell_date_str = `${year.toString()}-${month.toString().padStart(2, '0')}-${day.toString().padStart(2, '0')}`;
```

Comment:
can do `cell_date_str` = new Date(year, month, day).toString()` here and use additional date format options instead of just toString if you want more pretty printing if this gets displayed to the user